### PR TITLE
DDF-1348 Show error message instead of blank for nameless configs

### PR DESCRIPTION
**Pull request by brendan-hofmann**
[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-catalog/122)

### CODE REVIEW

**Comment by brendan-hofmann** [comment]
@AzGoalie @mverret @pklinef

**Comment by coyotesqrl** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/122)

File: admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js [hunk 1]
```diff
@@ -78,6 +78,9 @@ function (ich,Marionette,_,$,Q,ModalSource,EmptyView,Service,Status,wreqr,Utils,
             }
             data.available = this.model.get('available');
             data.name = this.model.get('name');
+            if (!data.name && data.currentConfiguration && data.currentConfiguration.id) {
```
Comment (andrew-fiedler) [inline admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: Any possibility of cleaning up the entire `serializeData` method?  It looks overcomplicated, and some of the logic just doesn't make sense.  

For instance, there is a check for whether `this.model` exists, but later on it is assumed to.

**Comment by mverret** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/122)

**Comment by mverret** [comment]
Full build completed successfully.  Tests ran successfully.  Manual verification was successful as well.

**Comment by mverret** [comment]
+1

**Comment by AzGoalie** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-catalog/122)

File: admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js (hunk 1 above)
Comment (brendan-hofmann) [inline admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: The real problem is that this logic exists at all. Overriding serializeData is a code smell. The real problem is the sloppy binding between the model and the view. That seems outside the scope of this ticket, I wouldn't mind a ticket to clean up this whole module though.

**Comment by pklinef** [comment]
+1

**Comment by andrew-fiedler** [comment]
:-1:  I'm thinking you should update ```sourceRow.handlebars ``` instead of updating ````Source.view.js````.

**Comment by brendan-hofmann** [comment]
I'm not sure why you say that. Conditional logic belongs more appropriately at the js level than the handlebars level. You *can* do it with templating, but I don't agree that that's the best way to do it.

File: admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js (hunk 1 above)
Comment (stustison) [inline admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: I don't agree at all that overriding serializeData is a code smell. That function exists specifically for the case where you want to present information to the template that you do not want to store inside your model. That being said, I agree with Andrew.

**Comment by stustison** [comment]
I don't think it is quite as black and white as that, to be honest. Conditional logic belongs wherever it makes the most sense. Sometimes, throwing the conditionals down into the template makes life far easier than trying to dump conditionals into the JS code or it simply overcomplicates the JS side. I say, do what makes sense and then let others throw stones. However, we went with handlebars instead of mustache in order to be able to use conditionals and other helpers that make life easier. So saying outright that "it is not the best way to do it" is just plain wrong, in my opinion.

**Comment by brendan-hofmann** [comment]
I should say that I don't think it's the best way to do it in this case. To my mind, handlebars conditionals are ideal when you want to alter the layout based on data in the model, that makes sense. However, this is just a string check for a single value, which seems to make more sense in the javascript. In other words, this doesn't seem like a conditional that belongs in the template. Like I said, really the model just be correct by the time it gets to the view.

File: admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js (hunk 1 above)
Comment (brendan-hofmann) [inline admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js]: That's not what's happening here though. The model has all the data, just not in the right place, and should instead map more accurately to the template. I don't think any of this logic belongs in the view, but rewriting the model seemed out of scope for this. I could still do that, there just might be a lot more side-effect changes.

**Comment by brendan-hofmann** [comment]
After further discussion with @stustison , we decided to show an error message if there is no name. The name is either the id or the shortname, and the catalog framework requires an id. If there is no id, then this is actually an invalid source configuration that can potentially cause other problems as well. Therefore, we should highlight the bad state instead of trying to hide it.

**Comment by shaundmorris** [comment]
:+1: on showing the error message. It could cause a lot of confusion as well problems (like you mentioned).
